### PR TITLE
Keysight B1500: Coerce nplc to integer

### PR DESCRIPTION
**Pull request by jdwatson**
Coerce coefficient for nplc settings to int.  B1500 does not update integration time if it receives a non-int value.

@astafan8

### CODE REVIEW

File: qcodes/instrument_drivers/Keysight/keysightb1500/KeysightB1500_base.py [hunk 1]
```diff
@@ -147,7 +147,7 @@ def _setup_integration_time(self,
                    )
 
     def use_nplc_for_high_speed_adc(
-            self, n: Optional[Union[int, float]] = None) -> None:
+            self, n: Optional[int = None) -> None:
```
Comment (jenshnielsen) [inline qcodes/instrument_drivers/Keysight/keysightb1500/KeysightB1500_base.py:150]:
```suggestion
            self, n: Optional[int] = None) -> None:
```

File: qcodes/instrument_drivers/Keysight/keysightb1500/KeysightB1500_base.py [hunk 2]
```diff
@@ -184,12 +185,14 @@ def use_nplc_for_high_resolution_adc(
...
+                to the B1500.
         """
         self._setup_integration_time(
             adc_type=constants.AIT.Type.HIGH_RESOLUTION,
             mode=constants.AIT.Mode.NPLC,
-            coeff=n
+            coeff=int(n)
         )
 
     def use_manual_mode_for_high_speed_adc(
```
Comment (jenshnielsen) [inline qcodes/instrument_drivers/Keysight/keysightb1500/KeysightB1500_base.py:198]: This probably needs the same change?

File: qcodes/instrument_drivers/Keysight/keysightb1500/KeysightB1500_base.py [hunk 3]
```diff
@@ -184,12 +185,14 @@ def use_nplc_for_high_resolution_adc(
...
-                (For more info see Table 4-21.)
+                (For more info see Table 4-21.).  Note that the integration
+                time will not be updated if a non-integer value is written
+                to the B1500.
         """
         self._setup_integration_time(
             adc_type=constants.AIT.Type.HIGH_RESOLUTION,
             mode=constants.AIT.Mode.NPLC,
-            coeff=n
+            coeff=int(n)
```
Comment (astafan8) [inline qcodes/instrument_drivers/Keysight/keysightb1500/KeysightB1500_base.py:195]: and for simplicity this conversion will be done in `_setup_integration_time`, not in every place where it's is called.

File: qcodes/instrument_drivers/Keysight/keysightb1500/KeysightB1500_base.py (hunk 2 above)
Comment (astafan8) [reply qcodes/instrument_drivers/Keysight/keysightb1500/KeysightB1500_base.py:200]: done

**Pull request merged**